### DISCUSSION TITLE
国内PT站tracker分流问题

### DISCUSSION THREAD

**Discussion by fishinsevens** [kkkgo/PaoPaoDNS#116] [问答和帮助] [answered]
目前PVE上LXC容器安装了2个PaoPaoDNS+虚拟机PaoPaoGateWay,使用中发现国内PT站tracker提示均为clash机场IP,请问是不是要手动把所有PT站tracker加入trackerslist.txt清单中?

**Comment by kkkgo** [comment, marked as answer]
是的，如果tracker比较明确数量不多的话添加到里面就好了。
如果下PT是专门的机器/虚拟机，可以直接在机器上配置手动不分流的DNS，如果是docker就更简单，直接docker选项就可以设置某个容器的DNS。

**Reply by fishinsevens** [reply]
感谢,已经手动把30个左右加进去了,目前ikuai还不支持手动设置单独的DNS,要等3.7.12正式版出来再修改一次就行了

**Comment by oneone1995** [comment]
发现加到trackerslist后对应的地址还是走了fakeip，CN_TRACKER和CNAUTO均是yes

**Reply by kkkgo** [reply]
可能使用的文本编辑器格式问题导致规则没生效，或者你填的trakcer格式不对。为了排除这个问题，你可以进入容器使用vi编辑器试试。

**Reply by oneone1995** [reply]
用vi重写了一遍还是没生效..我的tracker是`http://example.com/announce.php?passkey=xxxxx`。参考现在已有的例子，我加了个`http://example.com:80/announce`的一行记录。请教下，还能怎么定位么

**Reply by kkkgo** [reply]
不需要参考已有例子，正常tracker格式均可。
你可以开启ADDINFO功能使用dig命令查询结果的来源。
如果验证规则是否被容器正确处理，可以在容器内执行`grep example.com  /tmp/cn_tracker_list.txt`，如有full相关输出，并且没有其他规则冲突的话，`example.com`应该会被解析为原始DNS结果。

**Reply by oneone1995** [reply]
`grep example.com  /tmp/cn_tracker_list.txt`执行结果确实是`full:example.com`

dig的结果 192.168.88.185是paopaogw的地址
```

ash-4.4# dig example.com

; <<>> DiG 9.16.28 <<>> example.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 23995
;; flags: qr aa rd ra ad; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; MBZ: 0x0001, udp: 4096
; COOKIE: 6b43a151e323ee29 (echoed)
;; QUESTION SECTION:
;example.com.		IN	A

;; ANSWER SECTION:
example.com.	1	IN	A	11.0.0.165

;; Query time: 0 msec
;; SERVER: 192.168.88.185#53(192.168.88.185)
;; WHEN: Sat May 11 18:38:15 CST 2024
;; MSG SIZE  rcvd: 76
```

**Reply by kkkgo** [reply]
开启ADDINFO功能查看结果来源。

**Reply by kkkgo** [reply]
按你说的`192.168.88.185是paopaogw的地址`，你的拓扑可能存在问题，在任何情况下，除非你想要全局域名走paopaogw，否则不应该是paopaogw的地址作为DNS服务器。你的DNS规则是对PaoPaoDNS服务生效，跟paopaogw无关，paopaogw的DNS服务永远返回fakeip。

**Reply by oneone1995** [reply]
非常感谢！找到问题了
